test/Microsoft.AspNetCore.SignalR.Tests/HubConnectionHandlerTests.cs

```diff
@@ -2236,6 +2237,209 @@ public int Serialize(ref byte[] bytes, int offset, T value, IFormatterResolver f
+        {
+            var serviceProvider = HubConnectionHandlerTestUtils.CreateServiceProvider();
+
+            var connectionHandler = serviceProvider.GetService<HubConnectionHandler<MethodHub>>();
+
+            using (var client = new TestClient())
+            {
+                var connectionHandlerTask = await client.ConnectAsync(connectionHandler);
+                await client.BeginUploadStreamAsync("invocation", nameof(MethodHub.UploadArray), "id");
+
+                var objects = new[] { new SampleObject("solo", 322), new SampleObject("ggez", 3145) };
+
+                foreach (var thing in objects)
+                {
+                    await client.SendHubMessageAsync(new StreamItemMessage("id", thing));
+                }
+
+                await client.SendHubMessageAsync(new ChannelCompleteMessage("id"));
+                var response = (CompletionMessage)await client.ReadAsync();
+
+                var result = ((JArray)response.Result).ToArray<object>();
+
+                Assert.Equal(objects[0].Foo, ((JContainer)result[0])["foo"]);
+                Assert.Equal(objects[0].Bar, ((JContainer)result[0])["bar"]);
+                Assert.Equal(objects[1].Foo, ((JContainer)result[1])["foo"]);
+                Assert.Equal(objects[1].Bar, ((JContainer)result[1])["bar"]);
+            }
+        }
+
+        [Fact]
+        public async Task UploadManyStreams()
+        {
+            var serviceProvider = HubConnectionHandlerTestUtils.CreateServiceProvider();
+            var connectionHandler = serviceProvider.GetService<HubConnectionHandler<MethodHub>>();
+
+            using (var client = new TestClient())
+            {
+                var connectionHandlerTask = await client.ConnectAsync(connectionHandler);
+                var ids = new[] { "0", "1", "2" };
+
+                foreach (string id in ids)
+                {
+                    await client.BeginUploadStreamAsync("invocation_"+id, nameof(MethodHub.StreamingConcat), id);
+                }
+
+                var words = new[] { "karkalicious", "asdfsd", "ggez" };
+                var pos = new[] { 0, 0, 0 };
+                var order = new[] { 2, 2, 0, 2, 1, 0, 0, 0, 0, 0, 0, 2, 0, 1, 0, 1, 0, 0, 0, 1, 1, 1 };
+
+                foreach (var spot in order)
+                {
+                    await client.SendHubMessageAsync(new StreamItemMessage(spot.ToString(), words[spot][pos[spot]]));
+                    pos[spot] += 1;
+                }
+
+                foreach (string id in new[] { "0", "2", "1" })
+                {
+                    await client.SendHubMessageAsync(new ChannelCompleteMessage(id));
+                    var response = await client.ReadAsync();
+                    Debug.Write(response);
+                    Assert.Equal(words[Int32.Parse(id)], ((CompletionMessage)response).Result);
+                }
+            }
+        }
+
+        // NOTE -- THIS TEST DISPLAYS FUNKY BEHAVIOR
```

Comment:
This is known behavior in JSON.NET. It is extra generous with how it converts values. If you ask for a string but the input is a number, it'll stringify it. I don't think we need to have a big all caps comment here, maybe just a short note about the fact that JSON.NET is extra flexible here.